### PR TITLE
Form Validation on Login-Register View #89

### PR DESCRIPTION
**Pull request by hardikpatil**
In this PR:

- Added minor changes in form (As per docs/[Foodbank ER Diagram.png](https://github.com/ChicoState/PantryNode/blob/main/docs/Foodbank%20ER%20Diagram.png) schema , in the sign-up View, added fielda `First Name`, `Last Name` replacing `Name` field)
- Completed form validation on both login, and register views, with reference to issue #89

### CODE REVIEW

**Review by shmansa — APPROVED** [review]
Looks good to me!

Screenshot (250)
Screenshot (252)
Screenshot (254)
Screenshot (259)
Screenshot (260)
Screenshot (261)
Screenshot (262)

File: PantryNodeReact/src/pages/login.tsx [hunk 1]
```diff
@@ -13,21 +13,76 @@ import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
...
+    }
+    if (!/(?=.*[A-Z])(?=.*[\W_])/.test(passwordValue)) {
+      setPasswordError("Password must contain at least one number and one special character");
+      return;
+    }
+
     console.log({
       email: data.get('email'),
       password: data.get('password'),
     });
```
Comment (prasannarajezzzy) [inline PantryNodeReact/src/pages/login.tsx:80]: do we need to log user data ?

Comment (hardikpatil) [reply PantryNodeReact/src/pages/login.tsx:80]: Not really part of this PR but originally, I suppose this was for logging purpose to check what data to pass on other end points, but can remove this though, what do you say @prasannarajezzzy

Comment (prasannarajezzzy) [reply PantryNodeReact/src/pages/login.tsx]: Cool

**Review by prasannarajezzzy — APPROVED** [review]
Changes are working fine

**Comment by AbhinavReddy-Dev** [comment]
@hardikpatil @prasannarajezzzy can we please have the updated pages in the Figma for login and signup?

**Comment by hardikpatil** [comment]
> @hardikpatil @prasannarajezzzy can we please have the updated pages in the Figma for login and signup?

Oh yes, gotta update!

**Pull request merged**